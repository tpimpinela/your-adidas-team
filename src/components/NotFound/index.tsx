import { useCallback } from "react";
import { useNavigate } from "react-router-dom";
import Button from "../Button";

const NotFound = () => {
  const navigate = useNavigate();

  const handleOnClick = useCallback(() => {
    navigate("/");
  }, [navigate]);

  return (
    <>
      <h2>The specified route does not exist.</h2>
      <Button onClick={handleOnClick}>Go to the homepage</Button>
    </>
  );
};

export default NotFound;
