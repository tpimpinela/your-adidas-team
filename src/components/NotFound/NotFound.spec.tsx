import { render, RenderResult } from "@testing-library/react";
import { useNavigate } from "react-router-dom";
import { beforeEach, describe, expect, it, Mock, vi } from "vitest";
import NotFound from ".";
import userEvent from "@testing-library/user-event";

vi.mock("react-router-dom", () => ({
  useNavigate: vi.fn(),
}));

describe("<NotFound />", () => {
  let renderResult: RenderResult;
  let useNavigateMock: Mock<any, any>;

  beforeEach(() => {
    useNavigateMock = vi.fn();
    (useNavigate as Mock<any, any>).mockReturnValue(useNavigateMock);
    renderResult = render(<NotFound />);
  });

  it("should match the snapshot", () => {
    expect(renderResult.container).toMatchSnapshot();
  });

  it("should navigate to root route if button is clicked", async () => {
    await userEvent.click(renderResult.getByRole("button"));
    expect(useNavigateMock).toHaveBeenNthCalledWith(1, "/");
  });
});
