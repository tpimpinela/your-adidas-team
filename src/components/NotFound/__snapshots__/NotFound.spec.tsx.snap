// Vitest Snapshot v1

exports[`<NotFound /> > should match the snapshot 1`] = `
<div>
  <h2>
    The specified route does not exist.
  </h2>
  <button
    class="_button_2a1efe"
    data-testid="button-component"
  >
    Go to the homepage
  </button>
</div>
`;
